src/main/java/edu/kis/powp/jobs2d/command/ImmutableCompoundCommand.java

```diff
@@ -0,0 +1,44 @@
+package edu.kis.powp.jobs2d.command;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.visitor.CommandVisitor;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public final class ImmutableCompoundCommand implements ICompoundCommand {
+    private final List<DriverCommand> commands;
+    private final String name;
+
+    public ImmutableCompoundCommand(List<DriverCommand> commands, String name) {
+        this.commands = Collections.unmodifiableList(commands);
```

Comment:
commands within the list exist outside of ImmutableCompoundCommand and there is no guarantee that they will not be modified externally. We should make a deep copy of all commands (see PR #32). We also need to make sure, that references to any (possibly mutable) commands cannot be accessed from outside (see iterator() and getCommands() methods)